pkg/datapath/linux/probes/probes.go

```diff
@@ -422,3 +395,101 @@ func (p *ProbeManager) KernelConfigAvailable() bool {
 
 	return true
 }
+
+// HaveMapType is a wrapper around features.HaveMapType() to check if a certain
+// BPF map type is supported by the kernel.
+// On unexpected probe results this function will terminate with log.Fatal().
+func HaveMapType(mt ebpf.MapType) error {
+	err := features.HaveMapType(mt)
+	if errors.Is(err, ebpf.ErrNotSupported) {
+		return err
+	}
+	if err != nil {
+		log.WithError(err).Fatalf("failed to probe MapType: %s", mt.String())
+	}
+	return nil
+}
+
+// HaveProgramType is a wrapper around features.HaveProgramType() to check
+// if a certain BPF program type is supported by the kernel.
+// On unexpected probe results this function will terminate with log.Fatal().
+func HaveProgramType(pt ebpf.ProgramType) error {
+	err := features.HaveProgramType(pt)
+	if errors.Is(err, ebpf.ErrNotSupported) {
+		return err
+	}
+	if err != nil {
+		log.WithError(err).Fatalf("failed to probe ProgramType: %s", pt.String())
+	}
+	return nil
+}
+
+// HaveProgramHelper is a wrapper around features.HaveProgramHelper() to
+// check if a certain BPF program/helper copmbination is supported by the kernel.
+// On unexpected probe results this function will terminate with log.Fatal().
+func HaveProgramHelper(pt ebpf.ProgramType, helper asm.BuiltinFunc) error {
+	err := features.HaveProgramHelper(pt, helper)
+	if errors.Is(err, ebpf.ErrNotSupported) {
+		return err
+	}
+	if err != nil {
+		log.WithError(err).Fatalf("failed to probe Helper: %s/%s", pt.String(), helper.String())
+	}
+	return nil
+}
+
+// HaveLargeInstructionLimit is a wrapper around features.HaveLargeInstructions()
+// to check if the kernel supports the 1 Million instruction limit.
+// On unexpected probe results this function will terminate with log.Fatal().
+func HaveLargeInstructionLimit() error {
+	err := features.HaveLargeInstructions()
+	if errors.Is(err, ebpf.ErrNotSupported) {
+		return err
+	}
+	if err != nil {
+		log.WithError(err).Fatal("failed to probe large instruction limit")
```

Comment:
If you see the caller checking VTEP support for instance, you'll find that it's an optional feature. The caller already calls `log.Fatal(...)` but in that case the user gets the pointer that if they disable VTEP, they may still be able to successfully run Cilium. Fataling out here deprives the user of that knowledge.